compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -347,16 +347,16 @@ fn equate<'tcx>(
 #[instrument(level = "debug", skip(infcx))]
 fn negative_impl_exists<'tcx>(
     infcx: &InferCtxt<'tcx>,
-    o: &PredicateObligation<'tcx>,
+    obligation: &PredicateObligation<'tcx>,
     body_def_id: DefId,
 ) -> bool {
-    if resolve_negative_obligation(infcx.fork(), o, body_def_id) {
+    if resolve_negative_obligation(infcx.fork(), obligation, body_def_id) {
         return true;
     }
 
     // Try to prove a negative obligation exists for super predicates
-    for o in util::elaborate_predicates(infcx.tcx, iter::once(o.predicate)) {
-        if resolve_negative_obligation(infcx.fork(), &o, body_def_id) {
+    for elaborated in Elaborator::elaborate(infcx.tcx, obligation.clone()) {
```

Comment:
Passing in the predicate and then using the obligation that comes out implicitly sets the ParamEnv to empty... which I don't think is what we want? Maybe it doesn't matter though.